core/src/main/java/io/questdb/jit/CompiledFilterIRSerializer.java

```diff
@@ -580,13 +594,25 @@ private void serializeConstant(long offset, int position, final CharSequence tok
         }
 
         if (Chars.isQuoted(token)) {
-            if (PredicateType.CHAR != predicateContext.type) {
-                throw SqlException.position(position).put("char constant in non-char expression: ").put(token);
-            }
             if (len == 3) {
+                if (PredicateType.CHAR != predicateContext.type) {
+                    throw SqlException.position(position).put("char constant in non-char expression: ").put(token);
+                }
                 // this is 'x' - char
                 putOperand(offset, IMM, I2_TYPE, token.charAt(1));
                 return;
+            } else if (len == 2 + Uuid.UUID_LENGTH) {
+                if (PredicateType.UUID != predicateContext.type) {
+                    throw SqlException.position(position).put("uuid constant in non-uuid expression: ").put(token);
+                }
+                final CharSequence uuid = token.subSequence(1, token.length() - 1);
```

Comment:
can we avoid creating sub-sequence, this is GC right there. Perhaps move this logic to `Uuid.parseLo()`. Give it offset to parse from ?

Reply:
Fixed in: https://github.com/questdb/questdb/pull/3032/commits/3b7517222d2b3c645fe0345b2b8597448d4d8c93

Reply:
👍